java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/protocol/GetterSetterGenerator.java

```diff
@@ -18,58 +18,175 @@
+            result.add(createCodeAction(name, all ? CODE_GENERATOR_KIND : CodeActionKind.QuickFix, GENERATE_GETTERS_SETTERS, uri, offset, all, pair.first().stream().map(variableElement -> {
+                QuickPickItem item = new QuickPickItem(createLabel(info, variableElement));
+                item.setUserData(new ElementData(variableElement));
+                return item;
+            }).collect(Collectors.toList())));
+        }
+        return result;
+    }
+
+    @Override
+    public Set<String> getCommands() {
+        return commands;
+    }
 
+    @Override
+    @NbBundle.Messages({
+        "DN_SelectGetters=Select fields to generate getters for",
+        "DN_SelectSetters=Select fields to generate setters for",
+        "DN_SelectGettersSetters=Select fields to generate getters and setters for",
+    })
+    public CompletableFuture<Object> processCommand(NbCodeLanguageClient client, String command, List<Object> arguments) {
+        if (arguments.size() > 3) {
+            String uri = gson.fromJson(gson.toJson(arguments.get(0)), String.class);
+            int offset = gson.fromJson(gson.toJson(arguments.get(1)), Integer.class);
+            boolean all = gson.fromJson(gson.toJson(arguments.get(2)), boolean.class);
+            List<QuickPickItem> fields = Arrays.asList(gson.fromJson(gson.toJson(arguments.get(3)), QuickPickItem[].class));
+            int kind;
+            String text;
+            switch (command) {
+                case GENERATE_GETTERS: kind = GeneratorUtils.GETTERS_ONLY; text = Bundle.DN_SelectGetters(); break;
+                case GENERATE_SETTERS: kind = GeneratorUtils.SETTERS_ONLY; text = Bundle.DN_SelectSetters(); break;
+                default: kind = 0; text = Bundle.DN_SelectGettersSetters(); break;
+            }
+            if (all && fields.size() > 1) {
+                client.showQuickPick(new ShowQuickPickParams(text, true, fields)).thenAccept(selected -> {
+                    if (selected != null) {
+                        generate(client, kind, uri, offset, selected);
+                    }
+                });
+            } else {
+                generate(client, kind, uri, offset, fields);
+            }
+        }
+        return CompletableFuture.completedFuture(true);
+    }
+
+    private void generate(NbCodeLanguageClient client, int kind, String uri, int offset, List<QuickPickItem> fields) throws IllegalArgumentException {
+        try {
+            FileObject file = Utils.fromUri(uri);
+            JavaSource js = JavaSource.forFileObject(file);
+            List<TextEdit> edits = TextDocumentServiceImpl.modify2TextEdits(js, wc -> {
+                wc.toPhase(JavaSource.Phase.RESOLVED);
+                if (!fields.isEmpty()) {
+                    TreePath tp = wc.getTreeUtilities().pathFor(offset);
+                    tp = wc.getTreeUtilities().getPathElementOfKind(TreeUtilities.CLASS_TREE_KINDS, tp);
+                    if (tp != null) {
```

Comment:
I'd check the field's type TypeMirror for not being erroneous. Strange things may happen when generating incomplete types into getter / setter declartions.